packages/js/product-editor/src/components/manage-download-limits-modal/manage-download-limits-modal.tsx

```diff
@@ -0,0 +1,216 @@
+						'woocommerce'
+					),
+					DOWNLOAD_LIMIT_MIN
+				),
+			} ) );
+			return false;
+		}
+
+		setErrors( ( { downloadLimit: _, ...current } ) => current );
+		return true;
+	}
+
+	function validateDownloadExpiry() {
+		if (
+			downloadExpiry &&
+			! Number.isInteger( Number( downloadExpiry ) )
+		) {
+			setErrors( ( current ) => ( {
+				...current,
+				downloadExpiry: __(
+					'Expiry period must be an integer number',
+					'woocommerce'
+				),
+			} ) );
+			return false;
+		}
+
+		if ( Number.parseInt( downloadExpiry, 10 ) < 1 ) {
+			setErrors( ( current ) => ( {
+				...current,
+				downloadExpiry: sprintf(
+					__(
+						// translators: %d is the minimum value of the number input.
+						'Expiry period must be greater than or equal to %d',
+						'woocommerce'
+					),
+					DOWNLOAD_EXPIRY_MIN
+				),
+			} ) );
+			return false;
+		}
+
+		setErrors( ( { downloadExpiry: _, ...current } ) => current );
+		return true;
+	}
+
+	const downloadLimitProps = {
+		...useNumberInputProps( {
+			value: downloadLimit,
+			onChange: setDownloadLimit,
+		} ),
+		id: useInstanceId(
+			BaseControl,
+			'product_download_limit_field'
+		) as string,
+		type: 'number',
+		className: classNames( {
+			'has-error': errors.downloadLimit,
+		} ),
+		label: __( 'Download limit', 'woocommerce' ),
+		help:
+			errors.downloadLimit ||
+			__(
+				'Decide how many times customers can download files after purchasing the product. Leave blank for unlimited re-downloads.',
+				'woocommerce'
+			),
+		placeholder: __( 'Unlimited', 'woocommerce' ),
+		suffix: (
+			<span className="woocommerce-manage-download-limits-modal__input-suffix">
+				{ __( 'times', 'woocommerce' ) }
+			</span>
+		),
+		onBlur() {
+			validateDownloadLimit();
+		},
+	};
+
+	const downloadExpiryProps = {
+		...useNumberInputProps( {
+			value: downloadExpiry,
+			onChange: setDownloadExpiry,
+		} ),
+		id: useInstanceId(
+			BaseControl,
+			'product_download_expiry_field'
+		) as string,
+		type: 'number',
+		className: classNames( {
+			'has-error': errors.downloadExpiry,
+		} ),
+		label: __( 'Expiry period', 'woocommerce' ),
+		help:
+			errors.downloadExpiry ||
+			__(
+				'Decide how long customers can access the files after purchasing the product. Leave blank for unlimited access.',
+				'woocommerce'
+			),
+		placeholder: __( 'Unlimited', 'woocommerce' ),
+		suffix: (
+			<span className="woocommerce-manage-download-limits-modal__input-suffix">
+				{ __( 'days', 'woocommerce' ) }
+			</span>
+		),
+		onBlur() {
+			validateDownloadExpiry();
+		},
+	};
+
+	function handleSubmit( event: FormEvent< HTMLFormElement > ) {
+		event.preventDefault();
+
+		const isDownloadLimitValid = validateDownloadLimit();
+		const isDownloadExpiryValid = validateDownloadExpiry();
+
+		if ( isDownloadLimitValid && isDownloadExpiryValid ) {
+			onSubmit( {
+				downloadLimit: downloadLimit
+					? Number.parseInt( downloadLimit, 10 )
+					: null,
```

Comment:
Instead of `null` I think we want `-1`, same goes for `downloadExpiry`.
If we set this to `null` it saves it as zero, although `-1` is the default value, which I assume means unlimited.
I think you can also just set it as an empty string instead of specifically to `-1`

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/40797/commits/e5b244c93ba4a58516582a2806917b8349f34914